optimum/intel/utils/import_utils.py

```diff
@@ -196,8 +170,108 @@ def is_openvino_available():
+            ),
+        )
+    )
+
+
+@functools.lru_cache(1)
 def is_openvino_tokenizers_available():
-    return _openvino_tokenizers_available
+    if not is_openvino_available():
+        return False
+
+    try:
+        import openvino_tokenizers
+    except ImportError:
+        ov_major_version = _openvino_version.split("-")[0]
+        logger.info(
+            "OpenVINO Tokenizers is not available, tokenizer won't be converted. To automatically "
+            "convert tokenizer during OpenVINO model export install OpenVINO Tokenizers.\n"
+            "For PyPI distribution of OpenVINO:\n"
+            "pip install openvino-tokenizers[transformers]\n"
+            "For pre-release OpenVINO version:\n"
+            "pip install --pre -U openvino openvino-tokenizers"
+            "--extra-index-url https://storage.openvinotoolkit.org/simple/wheels/nightly\n"
+            "If you using archive OpenVINO distribution install release, nightly version or build from source:\n"
+            f"Release: pip install --no-deps openvino-tokenizers~={ov_major_version}\n"
+            f"Nightly: pip install --pre openvino-tokenizers~={ov_major_version}.0.dev --extra-index-url "
+            f"https://storage.openvinotoolkit.org/simple/wheels/nightly\n"
+            "Build from source instructions: https://github.com/openvinotoolkit/openvino_tokenizers"
+            "?tab=readme-ov-file#build-and-install-from-source\n"
+        )
+        return False
+
+    try:
+        pip_metadata_version = importlib_metadata.version("openvino")
+        is_nightly = False
+    except importlib_metadata.PackageNotFoundError:
+        try:
+            pip_metadata_version = importlib_metadata.version("openvino-nightly")
+            is_nightly = True
+        except importlib_metadata.PackageNotFoundError:
+            pip_metadata_version = False
+            is_nightly = False
+
+    try:
+        openvino_tokenizers._get_factory()
+    except RuntimeError as e:
+        tokenizers_version = openvino_tokenizers.__version__
+
+        if tokenizers_version == "0.0.0.0":
+            try:
+                tokenizers_version = importlib_metadata.version("openvino_tokenizers") or tokenizers_version
+            except importlib_metadata.PackageNotFoundError:
+                pass
+        message = (
+            "OpenVINO and OpenVINO Tokenizers versions are not binary compatible.\n"
+            f"OpenVINO version:            {_openvino_version}\n"
+            f"OpenVINO Tokenizers version: {tokenizers_version}\n"
+            "First 3 numbers should be the same. Update OpenVINO Tokenizers to compatible version. "
+        )
+        if not pip_metadata_version:
+            message += (
+                "For archive installation of OpenVINO try to build OpenVINO Tokenizers from source: "
+                "https://github.com/openvinotoolkit/openvino_tokenizers/tree/master?tab=readme-ov-file"
+                "#build-and-install-from-source"

```

Comment:
It would be good to explain which branch to pick, because building the source version only fixes a version conflict if the version the user has installed happens to be the same as the then current source version.

Reply:
This is not the case, the compatibility is defined by the OpenVINO version, not OpenVINO Tokenizers version. The only issue is if the archive OpenVINO version is old and OpenVINO Tokenizers use a newer API.

I can add a message about release branches in the OpenVINO Tokenizers repo instead.

Reply:
Update readme in [this PR](https://github.com/openvinotoolkit/openvino_tokenizers/pull/104)